src/Microsoft.Data.SqlClient/netcore/src/Microsoft/Data/SqlClientX/SqlConnector.cs

```diff
@@ -61,9 +63,15 @@ internal ValueTask Close(bool async)
         /// <param name = "cancellationToken">The token used to cancel an ongoing asynchronous call.</param>
         /// <returns>A Task indicating the result of the operation.</returns>
         /// <exception cref="NotImplementedException"></exception>
-        internal ValueTask Open(TimeSpan timeout, bool async, CancellationToken cancellationToken)
+        internal async ValueTask Open(TimeSpan timeout, bool async, CancellationToken cancellationToken)
         {
-            throw new NotImplementedException();
+            HandlerOrchestrator orchestrator = HandlerOrchestrator.Instance;
+            ConnectionHandlerContext context = new ConnectionHandlerContext
+            {
+                ConnectionString = new SqlConnectionString(DataSource.ConnectionString)
+            };
+            //TODO: a call to SqlCommandX once you reach end of chain.
+            await orchestrator.ProcessRequestAsync(context, async, cancellationToken).ConfigureAwait(false);         
```

Comment:
If we don't need to do anything else after initiating the request, we can return the ValueTask directly rather than awaiting it.
```suggestion
            return orchestrator.ProcessRequestAsync(context, async, cancellationToken);         
```

Reply:
+ remove the 'async' keyword from method.